src/audio/codec_adapter/codec_adapter.c

```diff
@@ -154,13 +154,15 @@ static int load_setup_config(struct comp_dev *dev, void *cfg, uint32_t size)
 		goto end;
 	}
 	/* Copy codec specific part */
-	lib_cfg = (char *)cfg + sizeof(struct ca_config);
 	lib_cfg_size = size - sizeof(struct ca_config);
-	ret = codec_load_config(dev, lib_cfg, lib_cfg_size, CODEC_CFG_SETUP);
-	if (ret) {
-		comp_err(dev, "load_setup_config(): %d: failed to load setup config for codec id %x",
-			 ret, cd->ca_config.codec_id);
-		goto end;
+	if (lib_cfg_size > 0) {
+		lib_cfg = (char *)cfg + sizeof(struct ca_config);
+		ret = codec_load_config(dev, lib_cfg, lib_cfg_size, CODEC_CFG_SETUP);
+		if (ret) {
+			comp_err(dev, "load_setup_config(): %d: failed to load setup config for codec id %x",
+				 ret, cd->ca_config.codec_id);
+			goto end;
+		}
```

Comment:
In fact, we should remove all goto end and replace with return ret. And return 0 at the end

Reply:
> In fact, we should remove all goto end and replace with return ret. And return 0 at the end

That should be another separated refinement if needed, this PR here only aim to fix the 0 codec config size issue.

Reply:
ok lets do that in a separate PR